server.go

```diff
@@ -2745,7 +2753,7 @@ func (ctx *RequestCtx) Done() <-chan struct{} {
 // RequestCtx.s.done is only closed when the server is shutting down
 func (ctx *RequestCtx) Err() error {
 	select {
-	case <-ctx.s.done:
+	case <-ctx.Done(): // fix  Use unified functions instead of reference variables to converge fetching into one place
```

Comment:
```suggestion
	case <-ctx.Done():
```
No need to add this as a comment.

Reply:
Already processed